scan.go

```diff
@@ -918,6 +911,32 @@ func stringToCm(token string) (e, m uint8, ok bool) {
 	return
 }
 
+func nameToAbsolute(name, origin string) (absolute string, err string) {
+	// origin reference (requires origin)
+	if name == "@" {
+		if origin == "" {
+			return "", "origin reference without origin"
```

Comment:
Yes, made possible by the change to start respecting empty origin instead of defaulting to the DNS root. So parsing the following edge-case zone files will now error for use of `@` without a defined origin:
```zone
; @ as owner name
@ 60 IN SOA ns.example. hostmaster 1 86400 60 86400 3600
```
```zone
example.com. 60 IN SOA ns.example. hostmaster 1 86400 60 86400 3600
example.com. 60      A 192.0.2.1
; @ in RDATA
1.2.0.192.in-addr.example.com. 60 PTR @
```

Reply:
ack, cool.